instrumentation/rails/lib/opentelemetry/instrumentation/rails/span_subscriber.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rails
+      class SpanSubscriber
+        def start(name, id, payload)
+          prev_ctx = OpenTelemetry::Context.current
+
+          span_name = name.split('.')[0..1].reverse.join(' ')
```

Comment:
It would be lovely to avoid these repeated allocations & string manipulation. I assume `name` does not vary much? Is it worth caching this mapping?

Reply:
I think it could be cached; and I also think there are some other small wins too (could cache the sorted listeners, as well).

I'll make those changes!